model-server/src/main/kotlin/org/modelix/model/server/store/IgniteStoreClient.kt

```diff
@@ -103,6 +112,43 @@ class IgniteStoreClient(jdbcConfFile: File? = null, inmemory: Boolean = false) :
         return cache.associate { it.key to it.value }
     }
 
+    override fun removeRepositoryObjects(repositoryId: RepositoryId) {
+        if (!inmemory) {
+            removeRepositoryObjectsFromDatabase(repositoryId)
+        }
+
+        val filter = IgniteBiPredicate<ObjectInRepository, String?> { key, _ ->
+            key.getRepositoryId() == repositoryId.id
+        }
+        val transformer = IgniteClosure<Cache.Entry<ObjectInRepository, String?>, ObjectInRepository?> { entry ->
+            entry.key
+        }
+        val query = ScanQuery(filter)
+        val toDelete = cache.query(query, transformer).all.asSequence().filterNotNull()
+            .toSortedSet() // sorting is necessary to avoid deadlocks
+        LOG.info { "Deleting cache entries asynchronously. count=${toDelete.size}" }
+        cache.removeAllAsync(toDelete).listen { LOG.info { "Cache entries deleted." } }
+    }
+
+    private fun removeRepositoryObjectsFromDatabase(repositoryId: RepositoryId) {
+        require(!inmemory) { "Cannot remove from database in inmemory mode." }
+        LOG.info { "Removing repository objects from database." }
+
+        dataSource.connection.use {
+            val statement = it.prepareStatement("DELETE from model WHERE repository = ?").apply {
+                setString(1, repositoryId.id)
+            }
+            try {
+                val deletedRows = statement.executeUpdate()
+                LOG.info { "Deleted rows from database. count=$deletedRows" }
```

Comment:
The meeting notes for the coding conventions workshop proposed this format: "Transaction failed [transaction=42]"

Reply:
done